python/pyarrow/tests/test_flight.py

```diff
@@ -1545,6 +1578,48 @@ def test_roundtrip_errors():
         with pytest.raises(flight.FlightInternalError, match=".*foo.*"):
             list(client.list_flights())
 
+        with pytest.raises(flight.FlightInternalError, match=".*foo.*"):
```

Comment:
Maybe we can put these errors in a list and loop over them, and then test both close() and write_table() separately? Something like

```python
exceptions = [flight.FlightInternalError, flight.FlightTimedOutError, ...]
for exception in exception:
    with pytest.raises(exception, match=...):
        writer, reader = client.do_put(...)
        writer.close()
    with pytest.raises(exception, match=...):
        writer, reader = client.do_put(...)
        writer.write_table(table)
        writer.close()
```

Reply:
it's a good idea.

Reply:
change it with a dict, and it can precisely control of exception occurres, but i don't know if not call write_table before close will cause other exception, so I test it with same code